### PR TITLE
[RF] Move internal enums of RooResolutionModels out of public interface

### PR DESCRIPTION
**Pull request by guitargeek**
This gives us more freedom when refactoring things to also support code
generation with AD.

Also apply other code modernizations, like less use of `TString` or
delegating constructors.

### CODE REVIEW

File: roofit/roofit/src/RooGExpModel.cxx [hunk 1]
```diff
@@ -321,49 +352,49 @@ double RooGExpModel::evaluate() const
     //double result = 0.5*evalCerf(fsign*u,c).real() ;  // sign=-1 !
 
     if (_basisCode!=0 && basisSign==Both) result *= 2 ;
-    //cout << "1st form " << "x= " << x << " result= " << result << endl;
+    //cout << "1st form " << "x= " << x << " result= " << result << std::endl;
```
Comment (lmoneta) [inline roofit/roofit/src/RooGExpModel.cxx:355]: This line could be conditional on `if (verboseEval()>...)` and similar for similar commented lines below ?

File: roofit/roofitcore/src/RooTruthModel.cxx [hunk 2]
```diff
@@ -31,17 +31,55 @@ functions used in D mixing have been hand coded for increased execution speed.
...
 #include <RooGenContext.h>
 
+#include <RooFit/Detail/EvaluateFuncs.h>
+
 #include <Riostream.h>
 
 #include <TError.h>
 
 #include <algorithm>
 #include <limits>
```
Comment (lmoneta) [inline roofit/roofitcore/src/RooTruthModel.cxx:42]: You might need to add here: `#include <cmath>` for std::exp and similar

**Review by lmoneta — APPROVED** [review]
LGTM
Just a couple of small comments

File: roofit/roofit/src/RooGExpModel.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/roofit/src/RooGExpModel.cxx:355]: Thanks for the suggestion, I will deal with this in the next refactoring.

**Pull request merged**